src/frontend/src/optimizer/rule/top_n_on_index_rule.rs

```diff
@@ -59,9 +59,17 @@ impl TopNOnIndexRule {
         let input_refs = scan_predicates.get_eq_const_input_refs();
         let prefix = input_refs
             .into_iter()
-            .map(|input_ref| ColumnOrder {
-                column_index: input_ref.index,
-                order_type: OrderType::ascending(),
+            .flat_map(|input_ref| {
+                vec![
+                    ColumnOrder {
+                        column_index: input_ref.index,
+                        order_type: OrderType::ascending(),
+                    },
+                    ColumnOrder {
+                        column_index: input_ref.index,
+                        order_type: OrderType::descending(),
+                    },
+                ]
```

Comment:
```suggestion
            .flat_map(|input_ref| {
                [
                    ColumnOrder {
                        column_index: input_ref.index,
                        order_type: OrderType::ascending(),
                    },
                    ColumnOrder {
                        column_index: input_ref.index,
                        order_type: OrderType::descending(),
                    },
                ]
```

Besides, `OrderType` has 4 variants, (ASC + DESC) * (nulls smallest + largest). For correctness, we may have to include all the 4 variants.